File: src/main/java/guru/xgm/language/angelscript/formatting/panel/AngelScriptCodeStyleMainPanel.java
package guru.xgm.language.angelscript.formatting.panel;

import com.intellij.application.options.TabbedLanguageCodeStylePanel;
import com.intellij.psi.codeStyle.CodeStyleSettings;
import com.intellij.psi.codeStyle.CodeStyleSettingsProvider;
import guru.xgm.language.angelscript.lang.AngelScriptLanguage;

public class AngelScriptCodeStyleMainPanel extends TabbedLanguageCodeStylePanel {

    public AngelScriptCodeStyleMainPanel(CodeStyleSettings currentSettings, CodeStyleSettings settings) {
        super(AngelScriptLanguage.INSTANCE, currentSettings, settings);
    }

    @Override
    protected void initTabs(CodeStyleSettings settings) {
        super.initTabs(settings);
        addTab(new AngelScriptAlignTokenPanel(settings));
        for (CodeStyleSettingsProvider provider : CodeStyleSettingsProvider.EXTENSION_POINT_NAME.getExtensionList()) {
            if (provider.getLanguage() == AngelScriptLanguage.INSTANCE && !provider.hasSettingsPage()) {
                createTab(provider);
            }
        }
    }
}

File: src/main/kotlin/raft/war/language/angelscript/formatting/panel/AngelScriptCodeStyleMainPanel.kt
package raft.war.language.angelscript.formatting.panel

import com.intellij.application.options.TabbedLanguageCodeStylePanel
import com.intellij.psi.codeStyle.CodeStyleSettings
import com.intellij.psi.codeStyle.CodeStyleSettingsProvider
import raft.war.language.angelscript.AngelScriptLanguage

class AngelScriptCodeStyleMainPanel(currentSettings: CodeStyleSettings?, settings: CodeStyleSettings?) :
    TabbedLanguageCodeStylePanel(AngelScriptLanguage.instance, currentSettings, settings!!) {
    override fun initTabs(settings: CodeStyleSettings) {
        super.initTabs(settings)
        addTab(AngelScriptAlignTokenPanel(settings))
        for (provider in CodeStyleSettingsProvider.EXTENSION_POINT_NAME.extensionList) {
            if (provider.language === AngelScriptLanguage.instance && !provider.hasSettingsPage()) {
                createTab(provider)
            }
        }
    }
}
